cccatalog-api/cccatalog/api/controllers/search_controller.py

```diff
@@ -53,23 +99,94 @@ def _quote_escape(query_string):
         return query_string
 
 
-def search(search_params, index, page_size, ip, page=1) -> Response:
+def _post_process_results(s, start, end, page_size, search_results,
```

Comment:
Moving this to `image_views` has broken a few features (slipped through some gaps in the test coverage).

- Thumbnail proxying is broken because THUMBNAIL, PROVIDER and URL, and THUMBNAIL_WIDTH_PX were not moved https://github.com/creativecommons/cccatalog-api/blob/d2e76395abacaef40c83b9d10979cf761bdbd8e8/cccatalog-api/cccatalog/api/controllers/search_controller.py#L134
- The endpoint `/image/related/<str:identifier>` is broken because it uses _post_process_results in the view but it has been moved to the search controller layer https://github.com/creativecommons/cccatalog-api/blob/d2e76395abacaef40c83b9d10979cf761bdbd8e8/cccatalog-api/cccatalog/api/views/image_views.py#L218

There might be some other similar issues elsewhere; I would suggest using an editor with type completion, highlighting, and refactoring to avoid issues like this (personally, I like Pycharm)

Reply:
Oops, I'm sorry about that. It should now be fixed.
I've also added one extra test just to call related images endpoint and check the response status and page size.

(In fact, I decided to give VSCode a try but didn't notice the code inspection wasn't well configured with the pipenv env and so warnings weren't popping up at all.)